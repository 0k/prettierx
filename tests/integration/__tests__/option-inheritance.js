"use strict";

const path = require("path");

const prettier = require("prettier-local");
const runPrettier = require("../runPrettier");

test("API resolveConfig with file arg should return only set options", () => {
  const file = path.resolve(
    path.join(__dirname, "../cli/config/option-inheritance/file.js")
  );
  return prettier.resolveConfig(file).then((result) => {
    expect(result).toMatchObject({
      bracketSpacing: true,
      objectCurlySpacing: false,
    });
  });
});

test("API resolveConfig.sync with file arg should return only set options", () => {
  const file = path.resolve(
    path.join(__dirname, "../cli/config/option-inheritance/file.js")
  );
  expect(prettier.resolveConfig.sync(file)).toMatchObject({
    bracketSpacing: true,
    objectCurlySpacing: false,
  });
});

describe("properly propogates inherited option values", () => {
  runPrettier("cli/config/option-inheritance/", ["./file.js"]).test({
    status: 0,
  });
});
