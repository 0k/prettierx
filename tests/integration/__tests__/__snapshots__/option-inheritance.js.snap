// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly propogates inherited option values (stderr) 1`] = `""`;

exports[`properly propogates inherited option values (stdout) 1`] = `
"import { name } from \\"package\\";

function optionInheritanceJs() {
  const obj = {x: 1};
  console.log(
    \\"option-inheritance/file.js should have no spacing in object brackets\\"
  );
  console.log(
    \\"option-inheritance/file.js should have spacing in import brackets\\"
  );
}
"
`;

exports[`properly propogates inherited option values (write) 1`] = `Array []`;
